pkg/util/helper/work_test.go

```diff
@@ -107,3 +130,165 @@ func TestGenEventRef(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkGetWorksByBindingIDWithoutIndex(b *testing.B) {
+	GetWorksByBindingID := func(ctx context.Context, c client.Client, bindingID string, namespaced bool) (*workv1alpha1.WorkList, error) {
+		var key string
+		if namespaced {
+			key = workv1alpha2.ResourceBindingPermanentIDLabel
+		} else {
+			key = workv1alpha2.ClusterResourceBindingPermanentIDLabel
+		}
+		workList := &workv1alpha1.WorkList{}
+		listOpt := &client.ListOptions{
+			LabelSelector: labels.SelectorFromSet(labels.Set{key: bindingID}),
+		}
+		return workList, c.List(ctx, workList, listOpt)
+	}
+	benchmarkGetWorksByBindingID(b, false, GetWorksByBindingID)
+}
+
+func BenchmarkGetWorksByBindingID(b *testing.B) {
+	benchmarkGetWorksByBindingID(b, true, GetWorksByBindingID)
+}
```

Comment:
Can you help to share the benchmark result here?

I tried to run it on my side but failed as I didn't prepare the CRDs and enviornment.

Reply:
I post the result in https://github.com/karmada-io/karmada/pull/5779#discussion_r1836746062 
With 50k works, the time taken by the function using index is reduced by 3000 times. Of course, this will have different results at other orders of magnitude, but it is a bit troublesome to test them one by one.